cmd/routed-eni-cni-plugin/cni.go

```diff
@@ -275,6 +277,38 @@ func add(args *skel.CmdArgs, cniTypes typeswrapper.CNITYPES, grpcClient grpcwrap
 	// dummy interface is appended to PrevResult for use during cleanup
 	result.Interfaces = append(result.Interfaces, dummyInterface)
 
+	if r.NetworkPolicyMode == "strict" {
+		// Set up a connection to the ipamD server.
+		npConn, err := grpcClient.Dial(npAgentAddress, grpc.WithTransportCredentials(insecure.NewCredentials()))
+		if err != nil {
+			log.Errorf("Failed to connect to network policy agent: %v", err)
+			return errors.Wrap(err, "add cmd: failed to connect to network policy agent backend server")
```

Comment:
At this point in time, we have already created a veth pair for the pod, configured ip rules, and marked the IP address in the datastore as in-use. Returning an error will cause the container-runtime to retry the CNI ADD, so are we sure that everything that has already been done is idempotent?

Note that above, if we fail to properly setup the virtual interfaces for the pod, we make a `c.DelNetwork` call before returning an error. I think that same recovery logic needs to apply here, plus maybe more.

Reply:
Yes, have a cleanup function to call..